### DISCUSSION TITLE
Flow X13 + XG Mobile

### DISCUSSION THREAD

**Discussion by Wolper321** [seerge/g-helper#265] [General]
Hi. 

First I want to thank you for your work. I have the Flow x13 (2022) without GPU (RA version) and the XGM 6850m XT.
Overall the performance modes work as expected: Silent 80W, Performance 100W and Turbo boosts XGM GPU to 165W.
In Armoury crate the second fan which supposes to be GPU called AUX. fan since the laptop doesn't have dgpu so I cannot tell if Ghelper detects the XGM fan at all. There should be 3 fans to control as in Armoury Crate (CPU + GPU (AUX) + XGM). 
The biggest issue so far is no control over the XGM activation/deactivation. It will be awesome if this will be implemented somehow in the future because for me it's the only thing stopping me from fully switching to Ghelper.

**Comment by seerge** [comment]
@Wolper321 hello, I don't own XGM mobile, so for me it's difficult to discover how to control it without physical device :) 
There is a topic for endpoint discovery https://github.com/seerge/g-helper/issues/38 where you can participate, mb you will spot endpoint responsible for enabling / disabling external GPU? :)

**Reply by seerge** [reply]
@Wolper321 try to run following in admin powershell (after physically connecting XGM)

```
Invoke-CimMethod (Get-CimInstance -Namespace root/wmi -ClassName AsusAtkWmi_WMNB) -MethodName DEVS -Arguments @{Device_ID=0x090019; Control_status=1}
```

1 to enable and 0 to disable?

**Reply by Wolper321** [reply]
You got this! This is the switch!
1 - ON and 0 - OFF. The connector correctly turns red and white when it's fine to disconnect.
The only problem present is that this is a brute-force solution.  The Armoury Crate is closing all the apps before connecting so Windows doesn't freak out when GPU is missing. Even Windows has the eject option but asks the user (instead of doing it by itself) to close the apps opened eGPU. I wonder if there is a small ASUS service (just for XGM plug in) I can keep without an Armoury Crate.

**Reply by seerge** [reply]
@Wolper321 

1. Before "disabling" XGM I can check for GPU usage (if its > 20% show confirmation warning), this is same as app does with regular dGPU switching. In general this is same situation as disabling dGPU (i.e. eco mode) that app does now. Killing all apps in this case is excessive as lots of apps that use GPU survive switching (like browsers / windows processes etc).

2. Eject option is a side effect of windows treating external gpu as usb drive or so :) it happens even when you swtich to standard from eco   :) that "eject" thing has no real effect

**Reply by Wolper321** [reply]
I also noticed that with the PowerShell switch browser didn't crash while usually, Armoury Crate was closing it. At least I switched it 2 or 3 times and Windows didn't freak out.

**Reply by seerge** [reply]
@Wolper321 ok, i can try to add a button in GPU section turning XGM on/off if it's connected (literally doing what command from powershell does).

Does you device support ultimate mode as well ?

**Reply by Wolper321** [reply]
Good question about the ultimate mode. My device doesn't have any modes since there is no dGPU. Probably permanent eco mode. But I am almost sure that there is a selection of modes for all the other people with dGPU (majority of z13, x13 and x16 users). So actually when XGM eGPU deactivates the internal dGPU should activate and I don't know how it will behave with the switch. 
Maybe some of the other users can test it.

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11273135/GHelper.zip)
You can try this build, i have added dedicated XGM button that will work as simple ON/OFF switch for XG Mobile
Button should be visible only if XGM is connected. 
App will check for "connection" every time it starts OR every time it's main window is being displayed (i.e. click on m4 or tray icon)

**Reply by Wolper321** [reply]
I have tried it and it is working as you described. When I fully unplug the device the XG mobile button disappeared and reappeared when I plug it back. Unfortunately, it didn't activate well, the red light appeared (it means that it's in use) but GPU didn't activate. After restarting everything started to work as expected. 
Happened to me once or twice when Armoury Crate asked me to restart for XGM to work. So I guess now the user should have some deduction abilities to use it :D Thank you!

**Reply by seerge** [reply]
@Wolper321 ok, cool. As for activation / reactivation - app just runs same command as you did manually in powershell. It could be that bios is acting here, but it's not much I can do here I think :)
